frontend/app_flowy/lib/plugins/grid/presentation/widgets/header/type_option/select_option.dart

```diff
@@ -273,6 +273,9 @@ class _CreateOptionTextField extends StatelessWidget {
                 .read<SelectOptionTypeOptionBloc>()
                 .add(SelectOptionTypeOptionEvent.createOption(optionName));
           },
+          onTap: () {
```

Comment:
Next, see if you can convert `_CreateOptionTextField` into a `StatefulWidget`, and allow passing in a `PopoverMutex`. Then, set up the `PopoverMutex` and `FocusNode` listeners. There's actually an extremely similar implementation [here](https://github.com/AppFlowy-IO/AppFlowy/blob/main/frontend/app_flowy/lib/plugins/grid/presentation/widgets/cell/date_cell/date_editor.dart#L298) that you can reference.